src/Altinn.AccessManagement.Integration/Clients/Altinn2RightsClient.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Altinn.AccessManagement.Core.Clients.Interfaces;
+using Altinn.AccessManagement.Core.Models;
+using Altinn.AccessManagement.Integration.Configuration;
+using Authorization.Platform.Authorization.Models;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Rest.Azure;
+
+namespace Altinn.AccessManagement.Integration.Clients
+{
+    /// <summary>
+    /// Client for getting Altinn roles from AltinnII SBL Bridge
+    /// </summary>
+    public class Altinn2RightsClient : IAltinn2RightsClient
+    {
+        private readonly SblBridgeSettings _sblBridgeSettings;
+        private readonly HttpClient _client;
+        private readonly ILogger _logger;
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AltinnRolesClient"/> class
+        /// </summary>
+        /// <param name="httpClient">HttpClient from default httpclientfactory</param>
+        /// <param name="sblBridgeSettings">the sbl bridge settings</param>
+        /// <param name="logger">the logger</param>
+        public Altinn2RightsClient(HttpClient httpClient, IOptions<SblBridgeSettings> sblBridgeSettings, ILogger<AltinnRolesClient> logger)
+        {
+            _sblBridgeSettings = sblBridgeSettings.Value;
+            _logger = logger;
+            _client = httpClient;
+            _serializerOptions.Converters.Add(new JsonStringEnumConverter());
+        }
+
+        /// <inheritdoc />
+        public async Task<DelegationCheckResponse> PostDelegationCheck(int userId,RightsDelegationCheckRequest request)
+        {
+            try
+            {
+                string partyId = request.From[0].Value;
```

Comment:
see comment on SingleRightsService. Method would be better to take these properties as individual input